packages/gen-ai-hub/src/orchestration/orchestration-client.ts

```diff
@@ -1,37 +1,45 @@
 import { executeRequest, CustomRequestConfig } from '@sap-ai-sdk/core';
 import { pickValueIgnoreCase } from '@sap-cloud-sdk/util';
-import { resolveDeployment } from '../utils/deployment-resolver.js';
+import {
+  DeploymentIdConfiguration,
+  resolveDeployment
+} from '../utils/deployment-resolver.js';
 import {
   CompletionPostRequest,
   CompletionPostResponse
 } from './client/api/schema/index.js';
-import { OrchestrationCompletionParameters } from './orchestration-types.js';
+import { OrchestrationModuleConfig, Prompt } from './orchestration-types.js';
 
 /**
  * Get the orchestration client.
  */
 export class OrchestrationClient {
+  // TODO: either use the deploymentId from here or only allow setting it through the request config path
+  // TODO: document constructor
+  constructor(
+    private config: OrchestrationModuleConfig,
+    private deploymentIdConfig?: DeploymentIdConfiguration
```

Comment:
I left the deployment ID config in here as it was already implemented below and required less changes, merely changed it to an object instead of string. If we decide to not allow setting it explicitly I will make a proposal without.

Reply:
For orchestration, in the future users will be able to store some parts of the config on server side and reference it by an ID. Accepting the deployment ID here would likely cause confusion once we add another opaque ID.

Personally, I'd remove the deployment ID it for now.

Reply:
Yes, I thought so, but I expect, that these references are part of the orchestration configuration. I don't really understand why you think it is *likely* to cause confusion, can you elaborate?

Reply:
Without knowing the details, I would assume you can set the deployment id also in the server side, which probably takes precedence, meaning you can select a deployment_id client side, but it will be ignored, which would be confusing.

Reply:
@MatKuhr is that what you meant? I thought by config we mean the orchestration config and it does not have any references to the deployment ID.

Reply:
> references are part of the orchestration configuration

That would be my expectation as well, but there is no decision on the details yet. I'd say it's not really problematic since the IDs will be in different objects, but still might increase the potential for user errors a little bit.